common/src/subgraph_client/client.rs

```diff
@@ -88,12 +88,23 @@ impl DeploymentDetails {
         graph_node_status_url: &str,
         graph_node_base_url: &str,
         deployment: DeploymentId,
+    ) -> Result<Self, anyhow::Error> {
+        Self::for_graph_node_url(
+            Url::parse(graph_node_status_url)?,
+            Url::parse(graph_node_base_url)?,
+            deployment,
+        )
+    }
+
+    pub fn for_graph_node_url(
+        graph_node_status_url: Url,
+        graph_node_base_url: Url,
```

Comment:
You don't want to pass those in as a struct so there's no mixing them up?

Reply:
There's a second refactor going on, I want to change as little as possible in common.